frame/staking-rewards/src/lib.rs

```diff
@@ -289,10 +326,43 @@ pub mod pallet {
 		#[transactional]
 		fn split(
 			_who: &Self::AccountId,
-			_position: &Self::PositionId,
-			_ratio: Permill,
+			position: &Self::PositionId,
+			ratio: Permill,
 		) -> Result<[Self::PositionId; 2], DispatchError> {
-			Err("Not implemented".into())
+			let mut old_position =
+				Stakes::<T>::try_mutate(position, |old_stake| match old_stake {
+					Some(stake) => {
+						let old_value = stake.clone();
+						stake.stake = ratio.mul_floor(stake.stake);
+						stake.share = ratio.mul_floor(stake.share);
+						let assets: Vec<T::AssetId> = stake.reductions.keys().cloned().collect();
+						for asset in assets {
+							let reduction = stake.reductions.get_mut(&asset);
+							if let Some(value) = reduction {
+								*value = ratio.mul_floor(*value);
+							}
+						}
+						Ok(old_value)
+					},
+					None => Err(Error::<T>::NoPositionFound),
+				})?;
+			let left_from_one_ratio = ratio.left_from_one();
+			let assets: Vec<T::AssetId> = old_position.reductions.keys().cloned().collect();
```

Comment:
I tried with https://paritytech.github.io/substrate/latest/frame_support/storage/bounded_btree_map/struct.BoundedBTreeMap.html#method.try_mutate but it consumes the map and I want in-place update so I used this.
Do you think any better way?

Reply:
You should be able to iterate over the old_position.reductions.keys() directly

Reply:
get_mut() requires mutable reference.
```rust
error[E0502]: cannot borrow `stake.reductions` as mutable because it is also borrowed as immutable
   --> frame/staking-rewards/src/lib.rs:341:24
    |
340 |                         for asset in stake.reductions.keys() {
    |                                      -----------------------
    |                                      |
    |                                      immutable borrow occurs here
    |                                      immutable borrow later used here
341 |                             let reduction = stake.reductions.get_mut(asset);
    |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here

```